src/borg/_hashindex.c

```diff
@@ -39,6 +40,18 @@ typedef struct {
     int32_t num_buckets;
     int8_t  key_size;
     int8_t  value_size;
+}) HashHeader1;
+
+BORG_PACKED(
+typedef struct {
+    char magic[MAGIC_LEN];
+    int32_t version;
+    int32_t num_entries;
+    int32_t num_buckets;
+    int32_t num_empty;
+    int32_t key_size;
+    int32_t value_size;
+    char reserved[1024 - 32];  // filler to 1024 bytes total
 }) HashHeader;
```

Comment:
Can we `_Static_assert(sizeof(HashHeader) == 1024, "hash header should be exactly 1024 bytes in size")`, or can't we rely on C11 support?

Reply:
i rarely do C programming these days, so not sure what we can rely on or what would be best to rely on.

borg 2 is expected to run on rather new systems (it also requires python 3.9), but supports a lot of platforms, thus the code should be portable.

Reply:
I suggest we add that, and if any compiler complains about lack of `_Static_assert`, fixing that can still be figured out? It's not like it'll cause any weird behaviour or breakage which is not detected ahead of shipping time.

Reply:
ok, I'll add it.